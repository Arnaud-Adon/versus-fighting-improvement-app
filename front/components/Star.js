import React from "react";
import { StyleSheet, Text, View, Dimensions } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { prefix } from "../utils/helper";

const { width } = Dimensions.get("window");

const Star = ({ title, number }) => {
  const { container, titleStyle, starContainer, starStyle } = styles;
  const renderStars = () => {
    switch (number) {
      case 1:
        return (
          <View testID="renderStars" style={starContainer}>
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
          </View>
        );
        break;

      case 2:
        return (
          <View testID="renderStars" style={starContainer}>
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
          </View>
        );
        break;
      case 3:
        return (
          <View testID="renderStars" style={starContainer}>
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
          </View>
        );
        break;

      case 4:
        return (
          <View testID="renderStars" style={starContainer}>
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star-outline`} style={starStyle} />
          </View>
        );
        break;

      case 5:
        return (
          <View testID="renderStars" style={starContainer}>
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
            <Ionicons name={`${prefix}-star`} style={starStyle} />
          </View>
        );
        break;
    }
  };
  return (
    <View style={container}>
      <Text style={titleStyle}>{title}</Text>
      {renderStars()}
    </View>
  );
};

export default Star;

const styles = StyleSheet.create({
  container: {
    alignContent: "center",
  },
  titleStyle: {
    fontWeight: "600",
    textAlign: "center",
  },
  starContainer: {
    marginVertical: 10,
    flexDirection: "row",
    justifyContent: "space-evenly",
  },
  starStyle: {
    fontSize: 30,
    color: "orange",
  },
});
