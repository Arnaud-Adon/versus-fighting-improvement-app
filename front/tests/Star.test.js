import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import AsyncStorage from "@react-native-community/async-storage";
import Star from "../components/Star";

describe("Component SelectCharacter", () => {
  beforeEach(() => {
    AsyncStorage.clear();
  });

  test("Le composant SelectCharacter est bien rendu", async () => {
    //Arrange
    const { toJSON } = render(<Star />);
    //Assert
    await waitFor(() => expect(toJSON()).toMatchSnapshot());
  });

  test('Que la vue renderStars exist', async () => {
    const { queryByTestId } = render(<Star />);

    await waitFor(() => {
      expect(queryByTestId("renderStars")).toBeTruthy();
    });
  });

  test('Si le rÃ©sulat contient au moins 5 Ionicons de type "star-outline"', () => {
    const { queryByTestId } = render(<Star />);

    await waitFor(() => {
      expect(queryByTestId("renderStars").props.children).toEqual(5);
    });
  })
});
