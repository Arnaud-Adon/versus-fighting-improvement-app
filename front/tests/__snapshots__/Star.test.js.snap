// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component SelectCharacter Le composant SelectCharacter est bien rendu 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "marginVertical": 10,
      }
    }
    testID="renderStars"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "orange",
            "fontSize": 30,
          },
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "orange",
            "fontSize": 30,
          },
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "orange",
            "fontSize": 30,
          },
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "orange",
            "fontSize": 30,
          },
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Object {
            "color": "orange",
            "fontSize": 30,
          },
          Object {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
